src/basic/parse-util.c

```diff
@@ -857,3 +857,28 @@ int parse_oom_score_adjust(const char *s, int *ret) {
         *ret = v;
         return 0;
 }
+
+int parse_loadavg_fixed_point(const char *s, unsigned long *ret) {
+        _cleanup_free_ char *a = NULL, *b = NULL;
+        unsigned long i = 0, f = 0;
+        int r;
+
+        assert(s);
+        assert(ret);
+
+        r = extract_many_words(&s, ".", 0, &a, &b, NULL);
+        if (r < 0)
+                return r;
+        if (r != 2)
+                return -EINVAL;
```

Comment:
i probably wouldn't use extract_many_words() here, all the fancy stuff it does, i.e. dealing with quoting, escaping and such is not useful here. Just use `strchr(s, '.')` to find the middle...

Also, i am pretty sure this isn't actually working correctly here, since trailing garbace isn't detected and refused. i.e. "1.2.3" would be parsable by this (the test cases really should cover that)